catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -0,0 +1,226 @@
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.content.operation.StorageRequest;
+import ddf.catalog.content.operation.UpdateStorageRequest;
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.content.operation.impl.UpdateStorageRequestImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.mime.MimeTypeMapper;
+import ddf.mime.MimeTypeResolutionException;
+
+public class ContentProducerDataAccessObject {
+
+    private static final transient Logger LOGGER =
+            LoggerFactory.getLogger(ContentProducerDataAccessObject.class);
+
+    public File getFileUsingRefKey(boolean storeRefKey, Message in)
+            throws ContentComponentException {
+        File ingestedFile;
+        try {
+            if (!storeRefKey) {
+                ingestedFile = ((GenericFile<File>) in.getBody()).getFile();
+            } else {
+                WatchEvent<Path> pathWatchEvent =
+                        (WatchEvent<Path>) ((GenericFileMessage) in).getGenericFile()
+                                .getFile();
+                ingestedFile = pathWatchEvent.context()
+                        .toFile();
+            }
+        } catch (ClassCastException e) {
+            LOGGER.warn(
+                    "Unable to cast message body to Camel GenericFile, so unable to process ingested file",
+                    e);
+            throw new ContentComponentException(
+                    "Unable to cast message body to Camel GenericFile, so unable to process ingested file");
+        }
+        return ingestedFile;
+    }
+
+    public WatchEvent.Kind<Path> getEventType(boolean storeRefKey, Message in) {
+        if (storeRefKey) {
+            WatchEvent<Path> pathWatchEvent =
+                    (WatchEvent<Path>) ((GenericFileMessage) in).getGenericFile()
+                            .getFile();
+            return pathWatchEvent.kind();
+        } else {
+            return StandardWatchEventKinds.ENTRY_CREATE;
+        }
+
+    }
+
+    public String getMimeType(ContentEndpoint endpoint, File ingestedFile)
+            throws ContentComponentException {
+        String fileExtension = FilenameUtils.getExtension(ingestedFile.getAbsolutePath());
+
+        String mimeType = null;
+
+        MimeTypeMapper mimeTypeMapper = endpoint.getComponent()
+                .getMimeTypeMapper();
+        if (mimeTypeMapper != null && ingestedFile.exists()) {
+            try (InputStream inputStream = Files.asByteSource(ingestedFile)
+                    .openStream()) {
+                if (fileExtension.equals("xml")) {
```

Comment:
Do you need to equalsIgnoreCase, or does the FilenameUtils class do a toLowercase on everything?